src/renderer/base/renderer.cpp

```diff
@@ -598,27 +598,21 @@ void Renderer::_PaintBufferOutputHelper(_In_ IRenderEngine* const pEngine,
     // If we have valid data, let's figure out how to draw it.
     if (it)
     {
-        // TODO: MSFT: 20961091 -  This is a perf issue. Instead of rebuilding this and allocing memory to hold the reinterpretation,
-        // we should have an iterator/view adapter for the rendering.
-        // That would probably also eliminate the RenderData needing to give us the entire TextBuffer as well...
-        // Retrieve the iterator for one line of information.
-        std::vector<Cluster> clusters;
         size_t cols = 0;
 
-        // Retrieve the first color.
-        auto color = it->TextAttr();
-
         // And hold the point where we should start drawing.
         auto screenPoint = target;
 
         // This outer loop will continue until we reach the end of the text we are trying to draw.
         while (it)
         {
+            TextBufferCellIterator runStartIt(it);

```

Comment:
When is inner loop is over `clusterIter` will be exhausted. So I copy it and keep the start of the run here.